From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shelley Vohr <shelley.vohr@gmail.com>
Date: Tue, 24 Aug 2021 21:50:26 +0200
Subject: fix: missing shell_dialog include

Fixes a build failure on Windows caused by chrome/browser/ui/browser_dialogs.cc
referencing ui::SelectedFileInfo and not including it. It's forward-declared
in chrome/browser/ui/browser_dialogs.h, so this adds the proper include
in the implementation.

Upstreamed at https://chromium-review.googlesource.com/c/chromium/src/+/3113977

diff --git a/chrome/browser/ui/browser_dialogs.cc b/chrome/browser/ui/browser_dialogs.cc
index 738a0c5602d8f6d672c569c8fa3618dd2c316e81..cd4a197300ff4cc1afe2bc339837d4d4355dc4a0 100644
--- a/chrome/browser/ui/browser_dialogs.cc
+++ b/chrome/browser/ui/browser_dialogs.cc
@@ -7,6 +7,9 @@
 #include "base/callback_helpers.h"
 #include "base/metrics/histogram_macros.h"
 
+#if !defined(TOOLKIT_VIEWS)
+#include "ui/shell_dialogs/selected_file_info.h"
+#endif
 namespace chrome {
 
 void RecordDialogCreation(DialogIdentifier identifier) {
